### ISSUE TITLE
Outside-government technology

### ISSUE THREAD

**Issue by vteague** [AusOpenTech/makingAPlan#4]
I’m prototyping an open-source system for allowing people to input important questions into parliamentary processes such as committee hearings, or just to ask their MP directly.  People will be able to suggest their own questions or upvote others’. The main challenge is making it usable for everyone while making sure that it protects the privacy of people’s political preferences.

What are you building?  What do you think needs to be built?  What are the main challenges to be addressed?

**Comment by KipCrossing** [comment]
Along with the [ausbills](https://github.com/OpenGovAus/Aus-Bills) project, **OpenGovAus** are also working on [legistream](https://github.com/OpenGovAus/legistream-backend) and it's corresponding [legistream-site](https://github.com/OpenGovAus/legistream-site) that are led by @king-millez. These projects aim to bring better access to Australian parliament live-streams.

We also have [many other projects in the works](https://github.com/OpenGovAus).

**Comment by king-millez** [comment]
> 
> 
> I’m prototyping an open-source system for allowing people to input important questions into parliamentary processes such as committee hearings, or just to ask their MP directly. People will be able to suggest their own questions or upvote others’. The main challenge is making it usable for everyone while making sure that it protects the privacy of people’s political preferences.
> 
> What are you building? What do you think needs to be built? What are the main challenges to be addressed?

As @KipCrossing said, we've been working on various projects as **OpenGovAus**, but expanding avenues with **AusOpenTech** could be something worth exploring.

**Comment by mukeshtiwari** [comment]
MiladKetabGhale, a student of Dirk Pattinson, has developed a modular framework for STV in Coq theorem prover which can be instantiate with different rules. You can see their code [1] . I have also developed a formally verified Schulze Method [2]  and a formally verified homomorphic Schulze Method [3]  in Coq theorem prover. 

What are the main challenges to be addressed?
I don't have an exact answer, but I have an idea/opinion that if a software program is used  by government for decision making, then it should be formally verified, similar to seL4, and open sourced.  This would guaranteed that at least software program is not garbage and it follows the specification. I would be very happy to discuss some more ideas which I have for strengthening the democracy of Australia. 



[1] https://github.com/MiladKetabGhale/Modular-STVCalculi
[2] https://github.com/mukeshtiwari/formalized-voting/tree/master/SchulzeOCaml
[3] https://github.com/mukeshtiwari/EncryptionSchulze/tree/master/code/Workingcode

**Comment by cofiem** [comment]
There are also the [OpenAustralia](https://github.com/openaustralia) [projects](https://www.oaf.org.au/projects/) - They vote for you, planning alerts, right to know. It would be good to pool resources, if possible, and not duplicate effort.

The [OpenAustralia Hansard Parser](https://github.com/openaustralia/openaustralia-parser) particularly might be useful to many of the suggestions in this repo.

**Comment by vteague** [comment]
Thankyou everyone, this is exactly the sort of useful list I was hoping to slowly get generated.  I completely agree that not duplicating effort is a key goal - indeed, just making sure that everyone working in this space has a fair idea of which other projects are running in parallel seems worthwhile.  (Did the Hansard Parser people and the Aus-Bills people already know about each other?)

With pooling resources and not duplicating effort in mind, I've set up a wiki with a list of projects [here](https://github.com/AusOpenTech/makingAPlan/wiki) and I'll set it up properly soon so others can add to it.

**Comment by cofiem** [comment]
Thanks for kicking this off, @vteague!

Some other related projects / research I'm aware of:
- [QUT Digital Media Research Centre's Australian Twitter News Index](https://datastudio.google.com/reporting/1vr1MWHDI91ja9nnxhd3ti3NI0lQWAJDK/page/Q4RO)
- [CiviWiki](https://github.com/CiviWiki/OpenCiviWiki) looked interesting when I last checked, but seems to be on hold at the moment.
- There's an [NZ govt law / Rules as Code movement](https://medium.com/@tom_5050/a-rules-as-code-case-study-from-new-zealand-92c7bbc74ba7). I'm not sure where this is at.
- This project has been on my radar as a way to index documents for research / journalism / finding connections [Aleph Data](https://github.com/alephdata/aleph).
